fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -285,77 +330,78 @@ private void validate(Map<String, String> properties) throws DdlException {
-                    return primitiveType == PrimitiveType.BIGINT;
-                }
-            case FLOAT:
-                return primitiveType == PrimitiveType.FLOAT;
-            case DOUBLE:
-                return primitiveType == PrimitiveType.DOUBLE;
-            case STRING:
-                return primitiveType == PrimitiveType.VARCHAR ||
-                        primitiveType == PrimitiveType.CHAR;
-            case ARRAY:
-                return validColumnType(avroSchema.getElementType(), ((ArrayType) srType).getItemType());
-            case FIXED:
-            case BYTES:
-                if (logicalType instanceof LogicalTypes.Decimal) {
-                    return primitiveType == PrimitiveType.DECIMALV2 || primitiveType == PrimitiveType.DECIMAL32 ||
-                            primitiveType == PrimitiveType.DECIMAL64 || primitiveType == PrimitiveType.DECIMAL128;
-                } else {
-                    return primitiveType == PrimitiveType.VARCHAR;
-                }
-            case UNION:
-                List<Schema> nonNullMembers = avroSchema.getTypes().stream()
-                        .filter(schema -> !Schema.Type.NULL.equals(schema.getType()))
-                        .collect(Collectors.toList());
+            Type convertSubType = null;
+            Type srSubType = null;
+            // considering ArrayLevel has recursions, we can get recursion levels here
+            int convertArrayLevel = 0;
+            int srArrayLevel = 0;
+            while (convertType instanceof ArrayType) {
+                convertSubType = ((ArrayType) convertType).getItemType();
+                convertType = convertSubType;
+                convertArrayLevel++;
+            }
+            while (srType instanceof ArrayType) {
+                srSubType = ((ArrayType) srType).getItemType();
+                srType = srSubType;
+                srArrayLevel++;
+            }
+            // the recursion levels should be same between convertType and srType
+            if (convertArrayLevel != srArrayLevel) {
+                valid = false;
+            }
+        }
 
-                if (nonNullMembers.size() == 1) {
-                    return validColumnType(nonNullMembers.get(0), srType);
-                } else {
-                    // UNION type is not supported in Starrocks
-                    return false;
-                }
-            case ENUM:
-            case MAP:
-            default:
-                return false;
+        // considering varchar info in hudi has no length, so we don't need to check length
+        // just check type here
+        if (convertType.isVarchar()) {
+            if (!srType.isVarchar()) {
+                valid = false;
+            }
+        } else {
+            // now, the complex types have been reduced to the simple primitive type
+            // we can compare two types by two types' toSql() here
+            valid = srType.toSql().equals(convertType.toSql());
         }
```

Comment:
As said in comments, varchar type got from avro meta has no length info, so if convertType or srType is varchar, we just need to varify their types, and no need to vertify their length.
And, till now, we can acquire the primitive type of convertType and srType.I come up with a idea that, we can use func `toSql()` to judge whether they are same.